lib/model/CForecastDataSink.cc

```diff
@@ -56,33 +56,19 @@ const std::string CForecastDataSink::STATUS("forecast_status");
 using TScopedAllocator = core::CScopedRapidJsonPoolAllocator<core::CRapidJsonConcurrentLineWriter>;
 
 CForecastDataSink::SForecastModelWrapper::SForecastModelWrapper(model_t::EFeature feature,
+                                                                const std::string& byFieldValue,
                                                                 TMathsModelPtr&& forecastModel,
-                                                                const std::string& byFieldValue)
-    : s_Feature(feature), s_ForecastModel(std::move(forecastModel)),
-      s_ByFieldValue(byFieldValue) {
-}
-
-CForecastDataSink::SForecastModelWrapper::SForecastModelWrapper(SForecastModelWrapper&& other)
-    : s_Feature(other.s_Feature), s_ForecastModel(std::move(other.s_ForecastModel)),
-      s_ByFieldValue(std::move(other.s_ByFieldValue)) {
+                                                                core_t::TTime firstDataTime,
+                                                                core_t::TTime lastDataTime)
+    : s_Feature(feature), s_FirstDataTime(firstDataTime), s_LastDataTime(lastDataTime),
+      s_ForecastModel(std::move(forecastModel)), s_ByFieldValue(byFieldValue) {
 }
 
 CForecastDataSink::SForecastResultSeries::SForecastResultSeries(const SModelParams& modelParams)
     : s_ModelParams(modelParams), s_DetectorIndex(), s_ToForecastPersisted(),
       s_ByFieldName(), s_MinimumSeasonalVarianceScale(0.0) {
 }
 
-CForecastDataSink::SForecastResultSeries::SForecastResultSeries(SForecastResultSeries&& other)
```

Comment:
be careful, this looks superfluous but is there for a reason: VS/Windows builds (though maybe only on 6.x). Would be good to run this change through `mybranch` prior commit

Reply:
Thanks, I'll check. I think if this works on master I'll leave with the default implementation and then adjust in the back port if necessary.